src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1845,29 +1845,81 @@ private FlagsExpression<FileAttributes> FormatAttributeSwitchParameters()
         /// Provides a mode property for FileSystemInfo.
         /// </summary>
         /// <param name="instance">Instance of PSObject wrapping a FileSystemInfo.</param>
+        /// <returns>A string representation of the FileAttributes, with one letter per attribute.</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods")]
-        public static string Mode(PSObject instance)
+        public static string Mode(PSObject instance) => Mode(instance, excludeHardLink: false);
+
+        /// <summary>
+        /// Provides a ModeWithoutHardLink property for FileSystemInfo, without HardLinks for performance reasons.
+        /// </summary>
+        /// <param name="instance">Instance of PSObject wrapping a FileSystemInfo.</param>
+        /// <returns>A string representation of the FileAttributes, with one letter per attribute.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods")]
+        public static string ModeWithoutHardLink(PSObject instance) => Mode(instance, excludeHardLink: true);
+
+        private static string Mode(PSObject instance, bool excludeHardLink)
         {
-            if (instance == null)
-                return string.Empty;
+            string ToModeString(FileSystemInfo fileSystemInfo)
+            {
+                FileAttributes fileAttributes = fileSystemInfo.Attributes;
+                bool isReparsePoint = InternalSymbolicLinkLinkCodeMethods.IsReparsePoint(fileSystemInfo);
+                bool isDirectory = fileAttributes.HasFlag(FileAttributes.Directory);
+                bool isHardLink = isReparsePoint || excludeHardLink ? false : InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileSystemInfo);
 
-            FileSystemInfo fileInfo = (FileSystemInfo)instance.BaseObject;
-            if (fileInfo == null)
-                return string.Empty;
+                var mode = new char[]
+                {
+                    isReparsePoint || isHardLink ? 'l' : isDirectory ? 'd' : '-',
+                    fileAttributes.HasFlag(FileAttributes.Archive)   ? 'a' : '-',
+                    fileAttributes.HasFlag(FileAttributes.ReadOnly)  ? 'r' : '-',
+                    fileAttributes.HasFlag(FileAttributes.Hidden)    ? 'h' : '-',
+                    fileAttributes.HasFlag(FileAttributes.System)    ? 's' : '-',
+                };
+                return new string(mode);
```

Comment:
I still don't understant why we need "return new"
```suggestion
                return mode;
```

Reply:
So that we return a string, and not a char[].

Reply:
Oh, I'm stupid :-) 
We have great new [String.Create()](https://docs.microsoft.com/en-us/dotnet/api/system.string.create?view=netcore-2.2) for the case.

Reply:
You can see the implementation here: https://github.com/dotnet/corefx/blob/a948042501969cf1968bec5fec998435c10e1bae/src/Common/src/CoreLib/System/String.cs#L332

Both do the same thing, but the ctor I'm using is doing it with less overhead.

Reply:
The temporary allocation on the stack, (the stackalloc) is just a part of the frame setup. You can consider it free. If you can show me that the string create version is faster, I'll be happy to change it, but I would be very surprised if that is the case.

Reply:
:-) Ah, yes, I spent a huge time on perf benchmark tests while I worked on Simple Case Folding implementation that it became a habit :-)

``` ini

BenchmarkDotNet=v0.11.3, OS=Windows 10.0.17763.253 (1809/October2018Update/Redstone5)
Intel Core i3-4150 CPU 3.50GHz (Haswell), 1 CPU, 4 logical and 2 physical cores
.NET Core SDK=2.1.403
  [Host]     : .NET Core 2.1.5 (CoreCLR 4.6.26919.02, CoreFX 4.6.26919.02), 64bit RyuJIT
  DefaultJob : .NET Core 2.1.5 (CoreCLR 4.6.26919.02, CoreFX 4.6.26919.02), 64bit RyuJIT


```
|       Method |     Mean |     Error |    StdDev | Ratio | RatioSD |
|------------- |---------:|----------:|----------:|------:|--------:|
| SpanToString | 18.92 ns | 0.2005 ns | 0.1675 ns |  1.00 |    0.00 |
| StringCreate | 11.82 ns | 0.3090 ns | 0.3306 ns |  0.63 |    0.02 |

String.Create() is very fast.
I expected it (why would Core team need to blow this method anyway?) but it's always better to check.

Reply:
> The temporary allocation on the stack, (the stackalloc) is just a part of the frame setup.

I'm not sure if this works for the used pattern with initiator.
For `ReadOnlySpan<char> mode = stackalloc char[4]` (use explicit constant) it works exactly.

Reply:
After looking into this quickly, it seems like string.Create might be slightly faster, but at the expense of creating a delegate. 
The delegate can be precreated and stored static, but then the clarity of the code is worse. 
The perf improvment is about 1e-8 seconds, which in my opinion is not worth it.

Reply:
My metering shows 38% win - Is this the purpose of this PR to make this method faster? :-)